notebooks/_06-Architecture.md

```diff
@@ -0,0 +1,429 @@
+ -  Class to estimate the choice model based on specified utility equations and retrieve model statistics
+
+## 5. Description of the minimally viable product (if any):
+- **Note the portions of the architecture responsible for producing the minimally viable product.**
+
+ - A minimally viable product is a notebook that takes data from outside sources and imports scripts including classes
+ from question 4, calls on the functions in each of the classes to simulate data, estimate models, and provide
+ parameters and statistics relevant to the problem at hand.
+
+## 6. Description of algorithms:
+### 6.1  What, if any, computational algorithms are being implemented as part of this project instead of relying upon external implementations?
+Hassan to fill in based on the deconfounder work.
+
+### 6.2 What purpose(s) do those algorithms serve?
+See above
+
+### 6.3 Are each of these algorithms fully understood by someone on the project team?
+See above
+
+### 6.4  What are alternative algorithms that could have served the purposes of the algorithms we are implementing?
+See above
+
+### 6.5 Why are we using the particular algorithms we’ve chosen instead of others? 
+See above
+
+### 6.6 Why do we believe the algorithms can be implemented?
+See above
+
+### 6.7 Do we have any concerns with regard to algorithmic efficiency?
+See above
+
+
+## 7. Description of data usage in the system:
+### 7.1 How will data be ingested / accessed?
+Data will be ingested using pandas from either an online path to the dataset or from a local path on the user's machine
+
+### 7.2 How will data be validated?
+Data can be validated using third party dependencies such
+
+### 7.3 How will data be organized?
+Data will be organized in tabular format (.csv)
+
+### 7.4 How will data be transformed?
+Data will be transformed using methods available within Pandas
+
+### 7.5 How will created data be output and stored?
+Causal graphs will be output to .png files and pushed to the repo
+
+Summaries of model estimate will be saved as markdown tables
+
+Plots of results will be saved as .png files and pushed to the repo
+
+### 7.6 How will created data be validated?
+There are packages that will validate data.
+
+### 7.7 How does all of the above differ online vs offline? (e.g. we may throw a critical error offline to prevent training with bad data but only log a warning online to note that the data in a request was unexpected.)
+N/A
+
+### 7.8 How will the flow / manipulation of data be controlled (and recorded / version-controlled) and made reproducible? E.g. data ingestion and manipulation to create training / testing sets.
+- **Flyte?**
+- **Make?**
+- **Other?**  
+We don't have any plans for this. please make any suggestions as you see needed/fit.
+
+## 8. Description of experimentation protocols (if relevant):
+**If experiments need to be run (e.g. hyper-parameter tuning, model-selection, A/B tests, etc.):**
```

Comment:
In this project, the experiments to be run are:
1. Changing a pre-specified causal graph from using independent X's to X's that cause other X's and comparing the change in the percentage of simulations where the data-generating parameters are recovered.
2. Changing one's estimation method from an MNL that doesn't account for unobserved confounding to an MNL model that does account for unobserved confounding.

In both cases, the general experiment may be re-run with different hyper-parameters. Imagine using two different distributions for the latent confounder that is excluded from one's graph. Or imagine using different causal graphs for the first case where some X's cause other X's.

In both scenarios, we may actually have to re-run the experiment multiple times.
- Joan may ask us to change the causal graph we're using.
- We might not get the results we're looking for in the first simulation and need to try differing distributions to make out point.
- Later reviewer's may dislike our chosen simulations and ask us to perform other ones (has happened to me in the past)

For all these reasons, we should set up an organized & reproducible way to run and track the experiments.